cpg-library/src/main/java/de/fraunhofer/aisec/cpg/meta/Edge.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2021, Fraunhofer AISEC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *                    $$$$$$\  $$$$$$$\   $$$$$$\
+ *                   $$  __$$\ $$  __$$\ $$  __$$\
+ *                   $$ /  \__|$$ |  $$ |$$ /  \__|
+ *                   $$ |      $$$$$$$  |$$ |$$$$\
+ *                   $$ |      $$  ____/ $$ |\_$$ |
+ *                   $$ |  $$\ $$ |      $$ |  $$ |
+ *                   \$$$$$   |$$ |      \$$$$$   |
+ *                    \______/ \__|       \______/
+ *
+ */
+package de.fraunhofer.aisec.cpg.meta;
+
+import de.fraunhofer.aisec.cpg.graph.Node;
+import java.util.Map;
+import java.util.Objects;
+
+public class Edge {
```

Comment:
That's more of a general view on edges that makes it straightforward to compare them. PropertyEdge is way more complex and isn't really suited for graph comparison that I need for my purposes. You can of course remove that from this PR as this is not relevant for concurrent frontends

Reply:
> That's more of a general view on edges that makes it straightforward to compare them. PropertyEdge is way more complex and isn't really suited for graph comparison that I need for my purposes. You can of course remove that from this PR as this is not relevant for concurrent frontends

Aaah, understand. Yes, so I think to not interfere with your thesis it would probably be good if you branch of this existing branch, maybe even before my merge? Then let's focus on concurrency here

Reply:
That's not an issue, I already branched off this one 👍